Sources/SwiftLanguageService/CodeActions/RemoveUnusedImports.swift

```diff
@@ -0,0 +1,278 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2025 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import BuildServerIntegration
+import Csourcekitd
+import Foundation
+package import LanguageServerProtocol
+import SKLogging
+import SourceKitD
+import SourceKitLSP
+import SwiftExtensions
+import SwiftSyntax
+
+package struct RemoveUnusedImportsCommand: SwiftCommand {
+  package static let identifier: String = "remove.unused.imports.command"
+  package var title: String = "Remove Unused Imports"
+
+  /// The text document related to the refactoring action.
+  package var textDocument: TextDocumentIdentifier
+
+  internal init(textDocument: TextDocumentIdentifier) {
+    self.textDocument = textDocument
+  }
+
+  package init?(fromLSPDictionary dictionary: [String: LanguageServerProtocol.LSPAny]) {
+    guard case .dictionary(let documentDict)? = dictionary[CodingKeys.textDocument.stringValue] else {
+      return nil
+    }
+    guard let textDocument = TextDocumentIdentifier(fromLSPDictionary: documentDict) else {
+      return nil
+    }
+
+    self.init(
+      textDocument: textDocument
+    )
+  }
+
+  package func encodeToLSPAny() -> LSPAny {
+    return .dictionary([
+      CodingKeys.textDocument.stringValue: textDocument.encodeToLSPAny()
+    ])
+  }
+}
+
+extension SwiftLanguageService {
+  func retrieveRemoveUnusedImportsCodeAction(_ request: CodeActionRequest) async throws -> [CodeAction] {
+    let snapshot = try await self.latestSnapshot(for: request.textDocument.uri)
+
+    let syntaxTree = await syntaxTreeManager.syntaxTree(for: snapshot)
+    guard
+      let node = SyntaxCodeActionScope(snapshot: snapshot, syntaxTree: syntaxTree, request: request)?
+        .innermostNodeContainingRange,
+      node.findParentOfSelf(ofType: ImportDeclSyntax.self, stoppingIf: { _ in false }) != nil
+    else {
+      // Only offer the remove unused imports code action on an import statement.
+      return []
+    }
+
+    guard let buildSettings = await self.compileCommand(for: request.textDocument.uri, fallbackAfterTimeout: true),
+      !buildSettings.isFallback,
+      buildSettings.compilerArgs.contains(["-enable-upcoming-feature", "MemberImportVisibility"])
+        || buildSettings.compilerArgs.contains(["-swift-version", "7"])
+    else {
+      // We can only offer the remove unused imports code action if `MemberImportVisibility` is enabled because
+      // otherwise imports of this source file might be necessary for other files to compile.
+      // We assume that `MemberImportVisibility` will be enabled in Swift version 7.
```

Comment:
Comment is technically incorrect given the discussion. If we're happy taking this change, IMO we don't need this check - it's basically "best effort".